projects/rocprofiler-systems/source/lib/rocprof-sys/library/amd_smi.cpp

```diff
@@ -241,6 +267,66 @@ get_state()
     static std::atomic<State> _v{ State::PreInit };
     return _v;
 }
+
+std::vector<uint8_t>
+serialize_xcp_metrics(const amdsmi_gpu_metrics_t& gpu_metrics)
+{
+    constexpr size_t vcn_count  = AMDSMI_MAX_NUM_VCN;
+    constexpr size_t jpeg_count = AMDSMI_MAX_NUM_JPEG;
```

Comment:
```suggestion
    constexpr uint8_t vcn_count  = AMDSMI_MAX_NUM_VCN;
    constexpr uint8_t jpeg_count = AMDSMI_MAX_NUM_JPEG;
```
Since you cast them to `uint8_t` right afterwards.